lib/sequent/core/command_service.rb

```diff
@@ -72,13 +72,13 @@ def execute_commands(*commands)
               @filters.each { |filter| filter.execute(command) }
 
               if command.valid?
-                @command_handlers.each do |command_handler|
-                  command_handler.handle_message command if command_handler.handles_message? command
-                end
+                parsed_command = command.parse_attrs_to_correct_types
+                @command_handlers.select { |h| h.handles_message?(parsed_command) }.each { |h| h.handle_message parsed_command }
+                @repository.commit(parsed_command)
+              else
+                raise CommandNotValid.new(command)
```

Comment:
you could also do early return and avoid an if/else:

`raise CommandNotValid.new(command) unless command.valid?` before the current if/else.